website/docs/d/ldap_dynamic_role_credentials.html.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: "vault"
+page_title: "Vault: vault_ldap_dynamic_credentials data source"
+sidebar_current: "docs-vault-datasource-ldap-dynamic_credentials"
+description: |-
+   Reads dynamic role credentials from an LDAP secret backend in Vault
+---
+
+# vault\_ldap\_dynamic\_credentials
+
+Reads dynamic role credentials from an LDAP secret backend in Vault
+
+~> **Important** All data retrieved from Vault will be
+written in cleartext to state file generated by Terraform, will appear in
+the console output when Terraform runs, and may be included in plan files
+if secrets are interpolated into any resource attributes.
+Protect these artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+## Example Usage
+
+```hcl
+resource "vault_ldap_secret_backend" "test" {
+  binddn                    = "..."
+  bindpass                  = "..."
+  url                       = "..."
+}
+
+resource "vault_ldap_secret_backend_dynamic_role" "role" {
+  mount         = vault_ldap_secret_backend.test.path
+  role_name     = "%s"
+  creation_ldif = <<EOT
+dn: cn={{.Username}},ou=users,dc=example,dc=org
+objectClass: person
+objectClass: top
+cn: learn
+sn: {{.Password | utf16le | base64}}
+userPassword: {{.Password}}
+EOT
```

Comment:
Max asked if there was a better way of handling this data other than HEREDOCs. I think the ldif attribute type "keys" can be custom. In that case, it might be possible to do something like [kvv2 data_json](https://github.com/hashicorp/terraform-provider-vault/blob/9f27afe07a3e13abb21003b005aa3ef8805353c2/vault/resource_kv_secret_v2.go#L89)

Reply:
right, I forgot about the base64 option. I am guessing using an LDIF file will be the most used scenario? In that case, I think we are good as-is.